Fix android profile download

nacochran opened the pull request:
This commit fixes two related bugs with Android MDM:

1. Android profiles now download correctly as .json files instead of .xml
   - Before: profiles downloaded as .xml with content '[object Object]'
   - After: profiles download as .json with properly formatted JSON content
   - Fixed by adding Android platform check in createProfileExtension() and createFileContent()

2. Custom Settings page now recognizes Android MDM
   - Before: showed 'MDM must be turned on' error even when Android MDM was enabled
   - After: properly detects Android MDM and allows profile management
   - Fixed by adding android_enabled_and_configured check to mdmEnabled
   - Backend middleware now supports Android MDM for profile endpoints
       a) Added VerifyAnyMDMConfigured() to support Apple, Windows, and Android MDM
       b) Updated profile endpoints to use VerifyAnyMDM() middleware

**Related issue:** Resolves #35023

# Checklist for submitter

- [x] Changes file added for user-visible changes in `changes/`
- [x] Input data is properly validated, `SELECT *` is avoided, SQL injection is prevented (using placeholders for values in statements)
- [x] If paths of existing endpoints are modified without backwards compatibility, checked the frontend/CLI for any necessary changes

## Testing

- [x] Added/updated automated tests
- [x] QA'd all new/changed functionality manually

## Database migrations

_No database migrations in this PR_

## New Fleet configuration settings

_No new Fleet configuration settings in this PR_

## fleetd/orbit/Fleet Desktop

_This PR does not affect fleetd/orbit/Fleet Desktop_

nacochran committed:
Fix Android profile download and Custom Settings page recognition

This commit fixes two related bugs with Android MDM:

1. Android profiles now download correctly as .json files instead of .xml
   - Before: profiles downloaded as .xml with content '[object Object]'
   - After: profiles download as .json with properly formatted JSON content
   - Fixed by adding Android platform check in createProfileExtension() and createFileContent()

2. Custom Settings page now recognizes Android MDM
   - Before: showed 'MDM must be turned on' error even when Android MDM was enabled
   - After: properly detects Android MDM and allows profile management
   - Fixed by adding android_enabled_and_configured check to mdmEnabled

3. Backend middleware now supports Android MDM for profile endpoints
   - Added VerifyAnyMDMConfigured() to support Apple, Windows, and Android MDM
   - Updated profile endpoints to use VerifyAnyMDM() middleware

Fixes #35023

nacochran committed:
Add changes file for Android profile download fix

nacochran committed:
Add mock implementation for VerifyAnyMDMConfigured method

MagnusHJensen commented on server/service/handler.go:
```diff
@@ -710,7 +710,7 @@ func attachFleetAPIRoutes(r *mux.Router, svc fleet.Service, config config.FleetC
 	mdmAppleMW.POST("/api/_version_/fleet/mdm/apple/profiles/match", matchMDMApplePreassignmentEndpoint, matchMDMApplePreassignmentRequest{})
 
-	mdmAnyMW := ue.WithCustomMiddleware(mdmConfiguredMiddleware.VerifyAppleOrWindowsMDM())
+	mdmAnyMW := ue.WithCustomMiddleware(mdmConfiguredMiddleware.VerifyAnyMDM())
```
I think it could be nice to add a super small integration tests, that hit one of these (`mdmAnyMW`) endpoints, without any MDM enabled, to see the `ErrMDMNotConfigured` error, but then enabling every MDM one by one, to ensure we don't see that specific error (shouldn't matter if the later logic fails).

I know it's not a complete solution, but having a single endpoint using the MW tested is better than nothing.

MagnusHJensen reviewed (changes requested):
Overall looks good! Again thank you.

I have two comments regarding adding some tests, while you address those I'll put the PR into draft, and once you've looked at it, just give me an @ ping.

nacochran committed:
Add tests and docs for Android-aware MDM middleware
   - add service-layer coverage for VerifyAnyMDMConfigured
   - extend middleware tests to cover Android flag and explain mock usage
   - add integration test hitting configuration profiles endpoint through new middleware
   - document service, middleware, and integration tests to clarify layering

nacochran committed:
Fix Android MDM middleware integration test setup

MagnusHJensen commented:
@nacochran Tests looks good, and thank you for the added docs to the methods.

I think it's a good callout, to remove the old `VerifyAppleOrWindowsMDM` especially since it's no longer used.
If you don't mind cleaning that up as well, then I think we are good to go with this PR.

nacochran committed:
Remove legacy Apple/Windows-only MDM checks

nacochran commented:
@MagnusHJensen The legacy code should be removed now! From what I looked at, I suspect some more simplifications could be made, but I don't want to get too deep into the woods with this PR. Hope this helps!

MagnusHJensen commented:
@nacochran Thanks, that looks great.

MagnusHJensen reviewed (approved):
LGTM, will ping someone with frontend approval rights to get this in.

iansltx reviewed (approved):
Approving for FE
